src/bedrock_agentcore/memory/integrations/strands/session_manager.py

```diff
@@ -536,40 +669,41 @@ def retrieve_for_namespace(namespace: str, retrieval_config: AgentCoreMemoryConf
                     except Exception as e:
                         # Continue processing other futures event if one fails rather than failing the entire operation
                         namespace = future_to_namespace[future]
-                        logger.error("Failed to retrieve memories for namespace %s: %s", namespace, e)
+                        logger.error(
+                            "Failed to retrieve memories for namespace %s: %s",
+                            namespace,
+                            e,
+                        )
 
-            # Inject customer context into the query
             if all_context:
+                original_text = messages[-1]["content"][0]["text"]
                 context_text = "\n".join(all_context)
-                ltm_msg: Message = {
-                    "role": "assistant",
-                    "content": [{"text": f"<user_context>{context_text}</user_context>"}],
-                }
-                event.agent.messages.append(ltm_msg)
-                logger.info("Retrieved %s customer context items", len(all_context))
+                messages[-1]["content"][0][
+                    "text"
+                ] = f"<previous_context>\n{context_text}\n</previous_context>\n\n{original_text}\n"
+                event.agent.messages[-1]["content"][0]["text"] = context_text
+                logger.info("Retrieved %d customer context items", len(all_context))
 
         except Exception as e:
             logger.error("Failed to retrieve customer context: %s", e)
 
     @override
     def register_hooks(self, registry: HookRegistry, **kwargs) -> None:
-        """Register additional hooks.
+        """Register hooks.
 
         Args:
             registry (HookRegistry): The hook registry to register callbacks with.
             **kwargs: Additional keyword arguments.
         """
-        RepositorySessionManager.register_hooks(self, registry, **kwargs)
```

Comment:
just to be clear, this line removes all the hooks that are expected as part of the session management approach in strands